src/topology/algebra/monoid.lean

```diff
@@ -149,6 +149,42 @@ section has_continuous_mul
 
 variables [topological_space M] [monoid M] [has_continuous_mul M]
 
+@[to_additive]
+lemma submonoid.closure_mul_self_subset (s : submonoid M) :
```

Comment:
I find this name confusing, as I'd expect it to be a lemma about `submonoid.closure` when in fact it's a lemma about `closure`. I think this would be less confusing as:
```suggestion
lemma closure_submonoid_mul_self_subset (s : submonoid M) :
```

Reply:
It would be sad to drop dot notation like that. What about `submonoid.top_closure_mul_self_subset` to avoid the ambiguity?

Reply:
Do you have an idea for a name that keeps the namespacing?

Reply:
I'm afraid I don't. @sgouezel's suggestion seems fair. This might be a naming problem worth asking zulip about - I haven't seen a case of needing to distinguish lemmas about `X.Y` and lemmas about `Y` on some `X` before, and whatever we decide would be good to have in the naming document.